pkg/framework/context/context.go

```diff
@@ -188,6 +199,14 @@ func WithProtocolServices(services ...dispatcher.Service) ProviderOption {
 	}
 }
 
+// WithGenericInboundServices injects services for handling generic inbound messages.
+func WithGenericInboundServices(services ...dispatcher.GenericInboundService) ProviderOption {
+	return func(opts *Provider) error {
+		opts.genericInboundServices = services
+		return nil
+	}
+}
+
```

Comment:
What is the larger design? We also need:
- exposed mechanism for external code / services to access a messenger (so they can send message, reply, etc).

I assume we will also want to plan:
- a dispatcher implementation of a messenger service that can handle dynamic registrations.
- REST binding of this dispatcher.
- REST binding for the messenger.

Reply:
- For sending message, still thinking about `OutboundHandler` for sending dynamic outbound message from handlers directly or having dedicated send message service.
- Yes, we need dynamic registration of dispatcher & REST bindings. These are still under design.

Reply:
You mean `Outbound` interface (or what is calling being called messenger) rather than `OutboundHandler`, correct?
https://github.com/hyperledger/aries-framework-go/blob/9da989cdfa11953add749c1959b2c8bce2630a80/pkg/didcomm/dispatcher/api.go#L21

Reply:
Yes